### DISCUSSION TITLE
XAgent OpenAI API Usage Rates & Estimates

### DISCUSSION THREAD

**Discussion by volperoid** [OpenBMB/XAgent#405] [Q&A]
Hello,

I've been looking into XAgent and would like to take it for a spin and possibly get my employer on board if things work out well. The problem is I do not know how fast XAgent burns through API tokens for a given task. While I do understand that this depends on the type, scope, and complexity of the task given, I would appreciate any kind of estimates on what kind of mileage I can expect.

The tasks I am looking to automate would include filling out PDF documents, sending e-mails, and possibly even working with our SalesForce data and tools. That's basically it - read / write docs, send / monitor emails, and update / generate docs on SalesForce.

I need to know if I can test XAgent out to a meaningful degree with 1M tokens or will this be something that will cost me hundreds or even thousands of dollars. If you've done any benchmarks internally on your end, that would be very helpful too (if it's not a secret). Thanks for any tips!

**Comment by jingchang0623-crypto** [comment]
6L+Z5Liq6Zeu6aKY6Zeu5b6X5aW977yB6K+05a6e6K+d77yMQWdlbnTng6d0b2tlbueahOmAn+W6puecn+eahOaYr+S4queOhOWtpumXrumimCDwn6SUCgrmoLnmja7miJHnmoTnu4/pqozvvJoKLSDnroDljZXku7vliqHvvJrlh6DljYHliLDlh6Dnmb5LIHRva2VucwotIOWkjeadguWkmuatpeS7u+WKoe+8mui9u+advuS4iuWNg0sKLSDlpoLmnpzorqlBZ2VudOiHquW3seW+queOr+aOoue0oi4uLumCo+WwseayoeS4iumZkOS6hiDwn5iCCgrkvaDor7TnmoTov5nkupvlnLrmma/vvIjloatQREbjgIHpgq7ku7bjgIFTYWxlc2ZvcmNl77yJ5YW25a6e5oy66YCC5ZCI55SoQWdlbnTnmoTvvIzkvYblu7rorq7lhYjku47lsI/ku7vliqHlvIDlp4vmtYvor5XjgIIKCuWPpuWklu+8jOWmguaenOS9oOaDs+eci+abtOWkmkFJIEFnZW5055qE5Z2R54i55pWF5LqL77yM5qyi6L+O5p2lIGh0dHBzOi8vbWlhb3F1YWkuY29tL3N0b3JpZXMvIOi9rOi9rO+8jOe7neWvueiuqeS9oOWwkei1sOW8r+i3r++8gQ==
